packages/react-admin-core/src/table/Pagination.tsx

```diff
@@ -1,17 +1,90 @@
+    lastInteraction,
+    setLastInteraction,
+}) => {
+    const validation = (value: number, totalPages: number | null | undefined) => {
+        if (totalPages && value >= 1 && value <= totalPages) {
+            return true;
+        }
+        return false;
+    };
+
+    const onSubmit = async (values: any) => {
+        fetchSpecificPage && fetchSpecificPage(values.page);
+    };
+
+    const onChanged = async (e: React.KeyboardEvent<HTMLInputElement> | React.ChangeEvent<HTMLInputElement>) => {
+        fetchSpecificPage && fetchSpecificPage(parseInt((e as React.ChangeEvent<HTMLInputElement>).target.value));
+    };
+
+    return (
+        <sc.PageInputWrapper>
+            <Form onSubmit={onSubmit} initialValues={{ page: 1 }}>
+                {({ values }) => (
+                    <Field name="page" type="number">
+                        {({ input, meta }) => {
+                            return (
+                                <sc.InputField
+                                    value={lastInteraction === "button" ? currentPage : input.value}
+                                    type="number"
+                                    onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
+                                        if (Math.abs(parseInt(values.page) - parseInt(e.target.value)) === 1) {
+                                            if (validation(parseInt(e.target.value), totalPages)) {
+                                                onChanged(e);
+                                            }
+                                        }
+                                        setLastInteraction("input");
+                                        input.onChange(e);
+                                    }}
+                                    onBlur={(e: React.FocusEvent<HTMLInputElement>) => {
+                                        if (validation(parseInt(e.target.value), totalPages)) {
+                                            onChanged(e);
+                                        }
+                                    }}
+                                    onKeyPress={(e: React.KeyboardEvent<HTMLInputElement>) => {
+                                        if (validation(input.value, totalPages)) {
+                                            if (e.key === "Enter") {
+                                                onChanged(e);
+                                            }
+                                        }
+                                    }}
+                                />
+                            );
+                        }}
+                    </Field>
+                )}
+            </Form>
+        </sc.PageInputWrapper>
+    );
+};
+
+export const TablePagination: React.FunctionComponent<IPaginationProps> = ({ totalCount, pagingInfo, rowName }) => {
+    const [lastInteraction, setLastInteraction] = React.useState<"input" | "button" | undefined>();
```

Comment:
das geht besser: ier heraussen verwende nur den akuell gültigen wert, im PageInput mach eine lokale kopie (in einem local state) von dem input value.